src/trace-plugin-interface.js

```diff
@@ -0,0 +1,246 @@
+  };
+}
+
+/**
+ * Creates a new Transaction object corresponding to an incoming request, which
+ * exposes methods operating on the root span.
+ * @param {object} options An object that specifies options for how the root
+ * span is created and propogated.
+ * @param {string} options.name The name to apply to the root span.
+ * @param {?string} options.url A URL associated with the root span, if
+ * applicable.
+ * @param {?function(string)} options.getHeader A function describing how to
+ * obtain a header field with the given field name from the incoming request
+ * assoicated with this root span. If supplied, it will be called to obtain the
+ * header field 'x-cloud-trace-context'.
+ * @param {?function(string, string)} options.setHeader A function describing
+ * how to set a header field with the given field name to a given value,
+ * respectively. If supplied, it will be called to set the header field in an
+ * outgoing request associated with this root span to the field name
+ * 'x-cloud-trace-context'.
+ * @param {?number} options.skipFrames The number of stack frames to skip when
+ * collecting call stack information for the root span, starting from the top;
+ * this should be set to avoid including frames in the PluginAPI. Defaults to 0.
+ * @returns A new Transaction object, or null if the trace agent's policy has
+ * disabled tracing for the given set of options.
+ */
+PluginAPI.prototype.createTransaction = function(options) {
+  options = options || {};
+  var that = this;
+  // If the options object passed in has the getHeader field set,
+  // try to retrieve the header field containing incoming trace metadata.
+  var incomingTraceContext;
+  if (is.fn(options.getHeader)) {
+    var header = options.getHeader('x-cloud-trace-context');
+    if (header) {
+      incomingTraceContext = that.agent.parseContextFromHeader(header);
+    }
+  }
+  incomingTraceContext = incomingTraceContext || {};
+  if (options.url && !that.agent.shouldTrace(options.url, incomingTraceContext.options)) {
+    return null;
+  }
+  var rootContext = that.agent.createRootSpanData(options.name,
+    incomingTraceContext.traceId,
+    incomingTraceContext.spanId,
+    options.skipFrames ? options.skipFrames + 1 : 1);
+  // If the options object passed in has the setHeader field set,
+  // use it to set trace metadata in an outgoing request.
+  if (is.fn(options.setHeader)) {
+    var outgoingTraceContext = rootContext.traceId + '/' +
+      rootContext.spanId;
+    var outgoingHeaderOptions = (incomingTraceContext.options !== null &&
+      incomingTraceContext.options !== undefined) ?
+      incomingTraceContext.options : constants.TRACE_OPTIONS_TRACE_ENABLED;
+    outgoingTraceContext += (';o=' + outgoingHeaderOptions);
+    options.setHeader('x-cloud-trace-context', outgoingTraceContext);
+  }
+  return new Transaction(that.agent, rootContext);
+};
+
+PluginAPI.prototype.getTransaction = function() {
+  if (cls.getRootContext()) {
```

Comment:
It does return null, but there still needs to be a branch, since we don't want `getTransaction` to return a `Transaction` object if there isn't one.